### ISSUE TITLE
mismatch between sum(mean(pop size)) and mean (metapop size)

### ISSUE THREAD

**Issue by mdahirel** [mdahirel/Tetranychus-metapop-2019#2]
following peer review, there were some criticisms of the stat model.
While most of these were not justified, some still pointed out to something that could be problematic or interesting, depending on how you look at it.

Basically, in the current version of the analysis, we generate the posterior for mean(metapopulation size) by summing the posteriors for mean(population size). This results in very precise means because our posteriors for patch-level means are very precise, which bugged the reviewer but isn't my point.

My point is that this actually may or may not be the right measure for mean metapopulation size (see https://doi.org/10.1016/j.tree.2020.10.007). (I'm not 100% sure yet, this paper may only apply to measures of patch-level K taken in isolated patches, which we haven't?)

A workaround would be to give up on sum(mean(pop size)) and actually focus on the patch level, at least for population size (we still would have alpha, beta and gamma variabilities at the metapop level).

But we could go further and maybe get even more interesting results! My idea is that we could add a model where we estimate the mean metapop size directly (we sum the patches **before** the model rather than after). By comparing the metapop sizes predicted by the "patch-level" model to the metapop sizes from the "metapop-level" model, we could have indirect info on dispersal and growth variation among patches and among treatments, which could be cool?

**Comment by mdahirel** [comment]
mmm, by reading theory, it looks like the comparison sum(mean(patch size)) vs mean(sum(patch size)) mentioned above is only relevant when we have access to mean(patch size) **in the absence of dispersal**, which we don't have. So not sure the comparison would be useful; will (a) look deeper and (b) still add a model for sum(patch size) in the code, just in case

edit: I mean we could assume that our low-connectedness treatment is closer to a no dispersal treatment, and that we could compare on these grounds, but not sure that it's a valid assumption

**Comment by dbonte** [comment]
Correct, that is for extrapolating. We are able to quantify it directly, so by summing. Hence, it is interesting to see differences in effective, realised metapop Ks.
Simply multiplying densities x the number of patches is a bad estimator because of potential differences in asynchrony (or in general the connectivity).

**Comment by mdahirel** [comment]
Oh yeah, I see.

So I guess in the end we would have something like this:
- patch-level population size: same interpretations as previous version, including the conclusions that were previously wrongly about metapops
- alpha, beta, gamma variabilities: same as previous
- metapopulation size (the "correct" one): probably no distinct effect given low sample size, but patterns in same direction as "multiplied patch-level" estimate
- comparison between "correct" metapop and "extrapolated multiplied patch-level" metapop : maybe informative if there are consistent differences, but not 100% sure about what yet

I *probably* will be able to redo the analyses before the 1st, so we'll have actual data to discuss this

**Comment by dbonte** [comment]
Yes! Sounds good

**Comment by mdahirel** [comment]
Closing this issue since I think I figured most of what to do in my "notes" script. everything seems to work fine based on tests; will implement it in the code proper monday
(important note: in the meantime, model postprocessing using the main code is broken, since the model part has already been updated but not the postprocessing part)

In brief:
- added an actual model for the metapopulation size
- changed the formula for the predicted patch population size. Previous version simply averaged over the individual replicate predictions; easy, but wouldn't generalise to new, unobserved replicates. Now using the closed-form expression for the mean on the observed scale, which should generalize; slightly wider CI, but conclusions unchanged for pop size;
- completely overhauled the postprocessing script; now use `purr` and should be less demanding in terms of memory (to implement to the code proper)

To do: 
- implement draft code to final code 
- for each plot, figure out (a) if there are observed data to plot along, (b) at what level we need to average the *observed* data to plot them along the predictions (it's a hierarchical model, so plotting raw data would not be illustrative of what's really happening).
- comment code
- double check conclusions are unchanged for alpha var (should be, based on prelim plots)
- compare 9*mean patch size vs. mean metapop size ?
